lib/tmuxinator/config.rb

```diff
@@ -123,20 +126,43 @@ def directories
         end
       end
 
+      def valid_project_config?(project_config)
+        return false unless project_config
+        unless exists?(path: project_config)
+          raise "Project config (#{project_config}) doesn't exist."
+        end
+        true
+      end
+
+      def valid_local_project?(name)
+        return false if name
+        raise NO_LOCAL_FILE_MSG unless local?
+        true
+      end
+
+      def valid_standard_project?(name)
+        return false unless name
+        raise "Project #{name} doesn't exist." unless exists?(name: name)
+        true
+      end
+
       def validate(options = {})
         name = options[:name]
         options[:force_attach] ||= false
         options[:force_detach] ||= false
-
-        project_file = if name.nil?
-                         raise NO_LOCAL_FILE_MSG \
-                           unless Tmuxinator::Config.local?
+        project_config = options.fetch(:project_config) { false }
+        project_file = if valid_project_config?(project_config)
```

Comment:
If I'm reading this right, `valid_project_config?` may raise an error, which is uncaught here and therefore needs to be handled by the caller of `validate`.  I don't really like that pattern, TBH, although we are [essentially already using it](https://github.com/tmuxinator/tmuxinator/blob/master/lib/tmuxinator/config.rb#L132)...  However, I would prefer if the error-raising was consolidated into one place, rather than distributed among multiple methods.  Maybe

```ruby
def valid_standard_project?(name)
  name ? true : false
end

def validate(options = {})
  valid = valid_project_config?(project_config) || valid_local_project?(name) || valid_standard_project?(name)

  if valid
    # now check/handle exception conditions
    raise "Project #{name} doesn't exist." unless exists?(name: name)
  
    # do other stuff?
  end
end
```

Thoughts?

Reply:
Yeah, I agree that those raises in the helper methods are less than Ideal. I had introduced the new conditional/raise alongside the others in the containing method and was going to open an issue in order to explore creating classes/structs to encapsulate the different project types and their validation behavior. However, in the meantime, Code Climate failed the build as a result of the new conditional and the simplest way forward was to move that validation logic into a method for each scenario.

I'll noodle around some more with your suggestion in mind and see if I can come up with a more clear, simpler solution.

Reply:
Gotcha.  Gotta love how CC sometimes pushes us to less-than-ideal solutions over something like that :)

If we think we're ok via test coverage for now, I'd say :shipit: and let's take a look at figuring out a better long-term solution.